### DISCUSSION TITLE
riverpod_generator define dependencies

### DISCUSSION THREAD

**Discussion by ArneSaknussemm89** [rrousselGit/riverpod#1821] [Q&A] [answered]
When using code generation, how would one define dependencies for the generated provider?

I have this (Firebase stuff):
```
@riverpod
FirebaseAuth firebaseAuth(FirebaseAuthRef ref) {
  final app = ref.watch(activeFirebaseAppProvider);
  return FirebaseAuth.instanceFor(app: app);
}
```
but the `activeFirebaseAppProvider` is a provider that is overridden with the actual FirebaseApp (codebase uses multiple Firebase projects for dev/stage/prod). So I need to define: `dependencies: [activeFirebaseAppProvider]` but I didn't see anywhere in the annotation to set those (my first guess).

Any thoughts?

**Comment by rrousselGit** [comment, marked as answer]
This is currently unsupported. There is a pending PR for it and it should be coming soon.

**Comment by ArneSaknussemm89** [comment]
Awesome! Looking forward to it! Loving the codegen so far.
